tools/esp8266/hmRadio.h

```diff
@@ -165,25 +165,37 @@ class HmRadio {
 
         void sendControlPacket(uint64_t invId, uint8_t cmd, uint16_t *data) {
             DPRINTLN(DBG_INFO, F("sendControlPacket cmd: ") + String(cmd));
-            sendCmdPacket(invId, TX_REQ_DEVCONTROL, ALL_FRAMES, false); // 0x80 implementation as original DTU code
+            sendCmdPacket(invId, TX_REQ_DEVCONTROL, SINGLE_FRAME, false); // 0x80 implementation as original DTU code???
             int cnt = 0;
-            mTxBuf[10] = cmd; // cmd --> 0x0b => Type_ActivePowerContr, 0 on, 1 off, 2 restart, 12 reactive power, 13 power factor
-            mTxBuf[10 + (++cnt)] = 0x00;
-            if (cmd >= ActivePowerContr && cmd <= PFSet){
-                mTxBuf[10 + (++cnt)] = ((data[0] * 10) >> 8) & 0xff; // power limit
-                mTxBuf[10 + (++cnt)] = ((data[0] * 10)     ) & 0xff; // power limit
-                mTxBuf[10 + (++cnt)] = ((data[1]     ) >> 8) & 0xff; // setting for persistens handlings
-                mTxBuf[10 + (++cnt)] = ((data[1]     )     ) & 0xff; // setting for persistens handling
+            mTxBuf[10] = cmd; // cmd --> 0x0b => Type_ActivePowerContr, 0 on, 1 off, 2 restart, 12 reactive power, 13 power factor, 20 CleanState_LockAndAlarm
+            if (cmd != CleanState_LockAndAlarm)
```

Comment:
I do not like this negative condition, it makes the code difficult to read.
Can you make it the other way round ?
```c
if (cmd == CleanState_LockAndAlarm) { // skip user data only crc8 }
else {
// append userdata and crc16
...
}
```

Reply:
We can change it, that make sense.
We need to change it then in the code, corrrect. Its not done yet? :)